### DISCUSSION TITLE
Collect statistics of a run and output to a file

### DISCUSSION THREAD

**Discussion by ocabra** [jesperpedersen/pgprtdbg#2] [Ideas]
Maybe pgprtdbg could have a command `stat` that collects and output statistics of a run to a file. I think it would make sense to have at least "No. of Transactions" and "Bytes Transferred". I could work on it.

**Comment by jesperpedersen** [comment]
Yeah, that could be an idea.

Maybe also having a format where each PostgreSQL message is printed in ASCII on one line could be an idea... Or at least expanding the `.out` format... that is a separate discussion.

Feel free to open an issue on it, and implement it

**Reply by ocabra** [reply]
I like that as well! Is the idea to basically structure what is currently outputed to console?

For example, the output to console is currenly:
```
--------
FE/Message (18):
51000000114449534341524420414C4C3B00
Q????DISCARD ALL;?
FE: Q
    Query: DISCARD ALL;
--------
BE/Message (23):
43000000104449534341524420414C4C005A0000000549
C????DISCARD ALL?Z????I
BE: C
    Tag: DISCARD ALL
BE: Z
    State: I
--------
FE/Message (35):
510000002253454C454354202A2046524F4D20706762656E63685F74656C6C65
727300
Q???"SELECT * FROM pgbench_telle
rs?
FE: Q
    Query: SELECT * FROM pgbench_tellers
--------
```
This is pretty detailed already. 

Is the idea to expand the messages to their meaningful name and save the output to a file?

**Reply by ocabra** [reply]
I think that less details could be just as helpful as well. Sometimes I don't need to have a dump of the data, I just want to see the message type.

**Reply by ocabra** [reply]
Maybe it makes more sense to design this first and then collect the statistics?

**Reply by jesperpedersen** [reply]
I meant have a `.bin` output where each packet is recorded in ASCII - one per line - , so like `<pid>-fe.bin`

```
51000000114449534341524420414C4C3B00
510000002253454C454354202A2046524F4D20706762656E63685F74656C6C65727300
```
